app/test/service.spec.ts

```diff
@@ -0,0 +1,109 @@
+import oasis from '@oasislabs/client';
+
+jest.setTimeout(20000);
+
+const GAS_LIMIT = '0xF42400';
+
+describe('Deployments', () => {
+  it ('should deploy non confidential', async () => {
+    let multisig = await oasis.workspace.Multisig.deploy(
+      ['b8b3666d8fea887d97ab54f571b8e5020c5c8b58'], 1,
+      {
+        header: {confidential: false},
+        gasLimit: GAS_LIMIT
+      }
+    );
+
+    var multisig_address = Buffer.from(multisig._inner.address).toString('hex');
+    let counter = await oasis.workspace.MultisigCounter.deploy(multisig_address, {
+      header: {confidential: false},
+      gasLimit: GAS_LIMIT
+    });
+
+    expect(multisig).toBeTruthy();
+    expect(counter).toBeTruthy();
+  });
+
+  it ('should deploy confidential', async () => {
+    let multisig = await oasis.workspace.Multisig.deploy(
+      ['b8b3666d8fea887d97ab54f571b8e5020c5c8b58'], 1,
+      { header: {confidential: true}, gasLimit: GAS_LIMIT}
+    );
+
+    var multisig_address = Buffer.from(multisig._inner.address).toString('hex');
+    let counter = await oasis.workspace.MultisigCounter.deploy(multisig_address, {
+      header: {confidential: true},
+      gasLimit: GAS_LIMIT
+    });
+
+    expect(multisig).toBeTruthy();
+    expect(counter).toBeTruthy();
+  });
+});
+
+describe('Test', () => {
+  let multisig, counter;
+
+  beforeAll(async () => {
+    multisig = await oasis.workspace.Multisig.deploy(
+      ['b8b3666d8fea887d97ab54f571b8e5020c5c8b58'], 1,
+      {
+        header: {confidential: false},
+        gasLimit: GAS_LIMIT
+      }
+    );
+
+    var multisig_address = Buffer.from(multisig._inner.address).toString('hex');
+    counter = await oasis.workspace.MultisigCounter.deploy(multisig_address, {
+      header: {confidential: false},
+      gasLimit: GAS_LIMIT
+    });
+  });
+
+  it('deployed', async () => {
+    expect(multisig).toBeTruthy();
+    expect(counter).toBeTruthy();
+  });
+
+  it('multisig getters work', async () => {
+    expect(await multisig.getRequired({gasLimit: GAS_LIMIT})).toEqual(1);
+    for (var func of counter._inner.idl.functions) {
+      if (func.name == 'increment') {
+        let coder = oasis.utils.OasisCoder.plaintext();
+        let txData = Array.from(await coder.encoder.encode(func, []));
+        let counter_address = Buffer.from(counter._inner.address).toString('hex');
+        expect(await multisig.addTransaction(counter._inner.address, 0, txData, {gasLimit: GAS_LIMIT})).toEqual(0);
+        expect(await multisig.isConfirmed(0, {gasLimit: GAS_LIMIT})).toEqual(false);
+        await multisig.confirmTransaction(0, {gasLimit: GAS_LIMIT});
+        expect(await multisig.isConfirmed(0, {gasLimit: GAS_LIMIT})).toEqual(true);
+        await multisig.revokeConfirmation(0, {gasLimit: GAS_LIMIT});
```

Comment:
eehhh do you really need to specify GAS_LIMIT everywhere? that's really sad, if so. like really sad

Reply:
Yes afaik, I do.